From 8611c5495af647759935f6d78670b5cae158f7e8 Mon Sep 17 00:00:00 2001
From: "Brett T. Warden" <brett.t.warden@intel.com>
Date: Wed, 5 Jul 2023 17:21:50 -0700
Subject: [PATCH] Add autoconf for dbus

---
 configure.ac | 55 ++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 55 insertions(+)

diff --git a/configure.ac b/configure.ac
index b358f9a1fa4a..0baa0eac959c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -150,6 +150,7 @@ dnl initialize all the info variables
     curl_ssl_msg="no      (--with-{openssl,gnutls,nss,mbedtls,wolfssl,schannel,secure-transport,amissl,bearssl,rustls} )"
     curl_ssh_msg="no      (--with-{libssh,libssh2})"
    curl_zlib_msg="no      (--with-zlib)"
+   curl_dbus_msg="enabled (--disable-dbus)"
  curl_brotli_msg="no      (--with-brotli)"
    curl_zstd_msg="no      (--with-zstd)"
     curl_gss_msg="no      (--with-gssapi)"
@@ -4370,6 +4371,59 @@ AS_HELP_STRING([--disable-websockets],[Disable WebSockets support]),
      AC_MSG_RESULT(no)
 )
 
+dnl D-Bus (required for autoproxy support)
+
+dnl Check for & handle argument to --with-dbus.
+
+clean_CPPFLAGS=$CPPFLAGS
+clean_LDFLAGS=$LDFLAGS
+clean_LIBS=$LIBS
+DBUS_LIBS=""
+AC_ARG_WITH(dbus,
+AS_HELP_STRING([--with-dbus=PATH],[search for dbus in PATH])
+AS_HELP_STRING([--without-dbus],[disable use of dbus]),
+               [OPT_DBUS="$withval"])
+
+if test "$OPT_DBUS" = "no" ; then
+    AC_MSG_WARN([dbus disabled])
+else
+  if test "$OPT_DBUS" = "yes" ; then
+    OPT_DBUS=""
+  fi
+
+  if test -z "$OPT_DBUS" ; then
+    CURL_CHECK_PKGCONFIG(dbus-1)
+
+    if test "$PKGCONFIG" != "no" ; then
+      LIBS="`$PKGCONFIG --libs-only-l dbus-1` $LIBS"
+      LDFLAGS="$LDFLAGS `$PKGCONFIG --libs-only-L dbus-1`"
+      CPPFLAGS="$CPPFLAGS `$PKGCONFIG --cflags-only-I dbus-1`"
+      OPT_DBUS=""
+    fi
+
+  fi
+
+  dnl Add a nonempty path to the compiler flags
+  if test -n "$OPT_DBUS"; then
+     CPPFLAGS="$CPPFLAGS -I$OPT_DBUS/include"
+     LDFLAGS="$LDFLAGS -L$OPT_DBUS/lib$libsuff"
+  fi
+
+  AC_CHECK_HEADER(dbus/dbus.h,
+    [
+    dnl dbus/dbus.h was found
+    HAVE_DBUS_H="1"
+    dnl if the lib wasn't found already, try again with the new paths
+    ],
+    [
+      dnl dbus/dbus.h was not found, restore the flags
+      CPPFLAGS=$clean_CPPFLAGS
+      LDFLAGS=$clean_LDFLAGS]
+    )
+fi
+
+dnl set variable for use in automakefile(s)
+AC_SUBST(DBUS_LIBS)
 
 dnl ************************************************************
 dnl hiding of library internal symbols
@@ -4733,6 +4787,7 @@ AC_MSG_NOTICE([Configured to build curl/libcurl:
   SSL:              ${curl_ssl_msg}
   SSH:              ${curl_ssh_msg}
   zlib:             ${curl_zlib_msg}
+  dbus:             ${curl_dbus_msg}
   brotli:           ${curl_brotli_msg}
   zstd:             ${curl_zstd_msg}
   GSS-API:          ${curl_gss_msg}
-- 
2.41.0

